esp/src/src/Utility.ts

```diff
@@ -273,6 +273,24 @@ export function isObjectEmpty(obj) {
     return true;
 }
 
+export function objectEquality(obj1, obj2) {
+    const obj1Props = Object.getOwnPropertyNames(obj1);
+    const obj2Props = Object.getOwnPropertyNames(obj2);
+
+    if (obj1Props.length !== obj2Props.length) {
+        return false;
+    }
+
+    for (let i = 0; i < obj1Props.length; ++i) {
+        const propName = obj1Props[i];
+
+        if (obj1[propName] !== obj2[propName]) {
+            return false;
+        }
+    }
+    return true;
+}
```

Comment:
This won't catch the following case:
```
obj1 ={
  a:1,
  b:2,
  c:3,
  e: undefined
}
obj2 ={
  a:1,
  b:2,
  c:3
  d:4
}
```

FYI @hpcc-js/util has a hashSum function which will create deep hashes of objects which you can then compare (which may not be what you want either).

Reply:
Can I compare equality of two objects with that hashSum? I need to make sure that they do no match to allow the insertion into the store otherwise keep out.

Reply:
Yes - but its a deep compare, so you need to be a bit careful as to what you pass in.,,

Reply:
`hashSum(obj1) === hashSum(obj2)`

Reply:
Our filter forms usually are as follows:

```
{
    WUID: "xxxx-abc",
    PublishedBy: "mvazquez",
}
```

They usually do not go any deeper than that.